### ISSUE TITLE
Possible tooling issue with Virtual Assistant Template and VS Enterprise 2019

### ISSUE THREAD

**Issue by managedcoder** [microsoft/botframework-solutions#1416]
## Project

**Name**: Virtual Assistant

**Language**: C#

## Description
Attempting to add Virtual Assistant Template solution to Git using Visual Studio 2019 Enterprise results in dialog that says "The current solution has projects that are located outside the solution folder. These projects will not be source controlled in the Git repository. To add all the projects to a single Git repository please consolidate all projects under a single folder."

## To Reproduce
Use the latest version of Visual Studio Enterprise 2019.  Install the latest Virtual Assitant and all its prerequisites.  Follow all the instructions for Getting Started with the Virtual Assistant Template (https://github.com/microsoft/botframework-solutions/blob/master/docs/tutorials/csharp/virtualassistant.md#download-and-install) and once the Solution is created in Visual Studio (first thing), right-click the VS Solution in the Solution Explorer and choose "Add Solution to Source Control" and you will see the error dialog.

## Expected behavior
It should have created a local Git repo without the menacing dialog popping up.

## Additional context
I checked and my colleague see the same behavior and it does not matter if “Place solution and project in the same directory” is checked or not on the File | New project dialog. I tried other project templates (ASP.NET MVC) and my installation of VS 2019 doesn’t have a problem with them when I do the same thing and add it to Git via VS 2019 (i.e. I don't see that dialog and it creates a local repo). Also, the Virtual Assistant Template works fine in VS 2017 and I can add it to Git with no issues which makes me wonder what is bothering the VS tooling about the VA template for VS 2019?  Something else that is strange is that, in spite of that ominous message, the local repo is actually created and the entire solution is added just like you would expect it to so not sure what is going on.

Does the VS Git tooling leverage anything from the generated solution and project files that the VA Template generates or could these things not possible be related?  Not sure why Git thinks the subfolders that have projects in them are outside the solution folder.

**Comment by jwiley84** [comment]
I saw this on stack overflow yesterday. I'm working on what's going wrong now.

**Comment by managedcoder** [comment]
Ok, sorry for posting twice but I haven’t been out to post on StackOverflow in a while and I was confused by the “This suggested edit to your post has already been handled and your action is no longer required” statement and the fact that there was no “reply” capability that I could see if I needed to reply.

Glad to hear you are on it.  I will check with other colleagues to see if its more common than just the one I checked with.

-Russ

**Comment by jwiley84** [comment]
I am unable to repro this:


![image](https://user-images.githubusercontent.com/17333688/58578601-08c03000-81fe-11e9-8fef-d0650ae24ceb.png)



What does your file structure look like?

**Comment by managedcoder** [comment]
It does not look like you were using VS 2019 to add to Git repo.  The problem (if there is a problem) I am trying to report is related to using the Git tooling in VS 2019  to add the repo to Git.  That is why in the repo steps I called that out.  I can see from your repo that Git commands work and that's good.

This may be a VS Git tooling issue but it seems strange that it only happens with this particular VA Template for VS 2019.  It does not happen with any other template in VS 2019 I try this with and it does not happen with the VA Template for VS 2017 when I am using VS 2017 so it makes me think something is different about the VA template for 2019 that could be changed so this issue does not happen.

**Comment by jwiley84** [comment]
Alright, let me try to repro exactly.

**Comment by jwiley84** [comment]
I was able to repro the issue you mentioned by following your steps EXACTLY. 

Investigating.

**Comment by jwiley84** [comment]
//sigh.

I was able to repro this issue just once. I got the error you mentioned only one time. Subsequent testing:

MyJARVIS was created inside VS 2019, with the 'Place solution and project in the same directory' box checked. I was able to successfully right click and add to source control:

![image](https://user-images.githubusercontent.com/17333688/58583800-ae2cd100-8209-11e9-81af-323e7feee7a9.png)

MyFRIDAY was created inside VS 2019 withOUT the box mentioned above checked. It also successfully was added to source control:

![image](https://user-images.githubusercontent.com/17333688/58583851-c56bbe80-8209-11e9-8ec7-54699fd7d69f.png)


Please check your .sln file and make sure both the main project and the .tests project are in the same relative pathing:

![image](https://user-images.githubusercontent.com/17333688/58584043-3612db00-820a-11e9-8a42-085b007dc05f.png)

**Comment by CoHealer** [comment]
@managedcoder, does this resolve your issue or you still need assistance?

**Comment by managedcoder** [comment]
This is a benign issue since the git repo is successfully created and works fine but I think this is still happening regularly with everyone I have worked with so far as we moved to the VA template and VS 2019.  Let me explain… I have to colleagues that started fresh with VS 2019 enterprise and the new VA template and each see this issue every time they create a new VA template bot and use VS 2019’s tooling to add the solution to git.  This past week I helped my customer to build their new VA template bot and they too saw this very same error.  I am not sure what’s different with your repo environment and ours… we are running Window 10 client (not Windows Server).

-Russ

**Comment by darrenj** [comment]
I have checked on two machines with VS2019 and can't repro - perhaps check the VS 2019 version? I am running 16.1.3 which seems to be fine.

**Comment by managedcoder** [comment]
So strange!  I am running the same version – 16.1.3.  I just did a bot workshop with a customer and both of their developers needed a fresh install of VS and then choose Community and both saw this strange dialog pop up once they used the tooling in lower right chrome of VS that says, “Add to Source Control”.  I see you closed it which is fine… just so weird… everyone one I worked with (internal and external) runs into this issue but no one on your side sees this the dialog.

[cid:image002.jpg@01D521C6.1531E3D0]
